test/vars/UtilSpec.groovy

```diff
@@ -435,4 +435,79 @@ class UtilSpec extends JenkinsPipelineSpecification {
         then:
         result == "/workspacefolder/kiegroup_projectA"
     }
+
+    def "[util.groovy] getNextVersionMicro"() {
+        when:
+        def snapshotVersion = groovyScript.getNextVersion('0.12.0', 'micro')
+        then:
+        snapshotVersion == '0.12.1-SNAPSHOT'
+        
+    }
+
+    def "[util.groovy] getNextVersionMinor"() {
+        when:
+        def snapshotVersion = groovyScript.getNextVersion('0.12.0', 'minor')
+        then:
+        snapshotVersion == '0.13.0-SNAPSHOT'
+        
+    }
+
+    def "[util.groovy] getNextVersionMajor"() {
+        when:
+        def snapshotVersion = groovyScript.getNextVersion('0.12.0', 'major')
+        then:
+        snapshotVersion == '1.12.0-SNAPSHOT'
+        
+    }
+
+    def "[util.groovy] getNextVersionContainsAlphabets"() {
+        when:
+        def checkForAlphabets = groovyScript.getNextVersion('a.12.0', 'micro')
+        then:
+        1 * getPipelineMock("error").call('Version is not in the required format.It should contain only numeric characters')
+
+        when:
+        checkForAlphabets = groovyScript.getNextVersion('a.12.0', 'minor')
+        then:
+        1 * getPipelineMock("error").call('Version is not in the required format.It should contain only numeric characters')
+        
+        when:
+        checkForAlphabets = groovyScript.getNextVersion('a.12.0', 'major')
+        then:
+        1 * getPipelineMock("error").call('Version is not in the required format.It should contain only numeric characters')
+    }
+    
+    def "[util.groovy] getNextVersionFormat"() {
+        when:
+        def checkForFormatError = groovyScript.getNextVersion('0.12.0.1', 'micro')
+        then:
+        1 * getPipelineMock("error").call('Version is not in the required format X.Y.Z')
+
+        when:
+        checkForFormatError = groovyScript.getNextVersion('0.12.0.1', 'minor')
+        then:
+        1 * getPipelineMock("error").call('Version is not in the required format X.Y.Z')
+
+        when:
+        checkForFormatError = groovyScript.getNextVersion('0.12.0.1', 'major')
+        then:
+        1 * getPipelineMock("error").call('Version is not in the required format X.Y.Z')
+    }
+
+    def "[util.groovy] getNextVersionAssertErrorCheck"() {
+        when :
+        groovyScript.getNextVersion('0.12.0', 'microo')
+        then:
+        thrown(Exception)
+
+        when :
+        groovyScript.getNextVersion('0.12.0', 'minoor')
+        then:
+        thrown(Exception) 
+
+        when :
+        groovyScript.getNextVersion('0.12.0', 'majjor')
+        then:
+        thrown(Exception)
+    } 
```

Comment:
The `suffix` argument is never tested. Please add a test for `groovyScript.getNextVersion('0.12.0', 'minor', 'whatever')` expecting `0.13.0-whatever` as a result